public/js/app.js

```diff
@@ -67,17 +68,42 @@ app.config(['$locationProvider',
     });
   }]);
 
-app.run(function (AuthService, annotationService, AUTH_EVENTS, $rootScope, $window, $location, $state, growl, SessionService) {
-  if(!(window.history && history.pushState)){
-      $rootScope.$on('$locationChangeStart', function(event) {
-          var loc = window.location;
-          var hash = loc.hash;
-          var path = loc.pathname || '/';
+app.run(function(AuthService, annotationService, AUTH_EVENTS, $rootScope,
+  $window, $location, $state, growl, SessionService, Page, $translate, $q) {
 
-          if(!hash){
-              loc.href = '/#' + path + (loc.search || '');
-          }
+  $rootScope.loadPages = function() {
+    return $q(function(resolve, reject) {
+
+      Page.query(function(pages) {
```

Comment:
If I'm reading this correctly, you're getting the entire content of every page on the site here.  That's rather slow and expensive, perhaps adding a `format=no-content` or similar parameter would be good?